### PR TITLE
Bump bash-completion to 2.8, leverage GNU grep, awk, sed where possible

### PR DESCRIPTION
**Pull request by Mno-hime**
Fixed patches/01-ggrep.patch to apply to 2.8 and leverage GNU awk and
GNU sed in the completions to fix various problems in completion where
bash-completion generally expects GNU tools.

Test: Typed the command for completion and pressed `tab` once or twice.

Old:

```
{build-userland} root:~/ws/oi-userland/components/shell/bash-completion # vncviewer awk: syntax error near line 1
awk: bailing out near line 1
awk: syntax error near line 1
awk: bailing out near line 1

::1        localhost  loghost

{build-userland} root:~/ws/oi-userland/components/shell/bash-completion # gmake "    1,/^# * Make data b ...": 22: RE error: invalid regular expression
```

New:

```
{build-userland} root:~/ws/oi-userland/components/shell/bash-completion # vncviewer
192.168.1.92       192.30.253.112     ::1                github.com         localhost          loghost            toshiba.mnowak.cz

{build-userland} root:~/ws/oi-userland/components/shell/bash-completion # gmake
clean     download  patch     unpack
```

`sample-manifest` target changed couple of lines.

### CODE REVIEW

File: components/shell/bash-completion/patches/02-gawk.patch [hunk 1]
```diff
@@ -0,0 +1,1254 @@
...
+-            local packages_list=$(awk ' $0 ~ /BEGIN PGP SIGNATURE/ { exit } $1 ~ /^Hash:/ || $1 ~ /^ *(-|#|$)/ { next } { print $1 }' "${catalog_files[@]}")
++            local packages_list=$(gawk ' $0 ~ /BEGIN PGP SIGNATURE/ { exit } $1 ~ /^Hash:/ || $1 ~ /^ *(-|#|$)/ { next } { print $1 }' "${catalog_files[@]}")
+             COMPREPLY=( $(compgen -W "${packages_list}" -- $cur) )
+ 
+         elif [[ "$command" == @(-r|--remove) ]]; then
+-                local packages_list=$(pkginfo | awk ' $2 ~ /^CSW/ { printf ("%s|",$2) }')
++                local packages_list=$(pkginfo | gawk ' $2 ~ /^CSW/ { printf ("%s|",$2) }')
+             packages_list=${packages_list%|}
+-            packages_list=$(nawk " \$3 ~ /^$packages_list\$/ { print \$1 }" "${catalog_files[@]}")
++            packages_list=$(ngawk " \$3 ~ /^$packages_list\$/ { print \$1 }" "${catalog_files[@]}")
```
Comment (pyhalov) [inline components/shell/bash-completion/patches/02-gawk.patch]: we don't have "ngawk"

File: components/shell/bash-completion/patches/02-gawk.patch [hunk 2]
```diff
@@ -0,0 +1,1254 @@
...
+-        tmpkh=( $( awk 'sub("^[ \t]*([Gg][Ll][Oo][Bb][Aa][Ll]|[Uu][Ss][Ee][Rr])[Kk][Nn][Oo][Ww][Nn][Hh][Oo][Ss][Tt][Ss][Ff][Ii][Ll][Ee][ \t]+", "") { print $0 }' "${config[@]}" | sort -u ) )
++        tmpkh=( $( gawk 'sub("^[ \t]*([Gg][Ll][Oo][Bb][Aa][Ll]|[Uu][Ss][Ee][Rr])[Kk][Nn][Oo][Ww][Nn][Hh][Oo][Ss][Tt][Ss][Ff][Ii][Ll][Ee][ \t]+", "") { print $0 }' "${config[@]}" | sort -u ) )
+         IFS=$OIFS
+         for i in "${tmpkh[@]}"; do
+             # First deal with quoted entries...
+@@ -1561,34 +1561,34 @@ _known_hosts_real()
+ 
+     # If we have known_hosts files to use
+     if [[ ${#kh[@]} -gt 0 || ${#khd[@]} -gt 0 ]]; then
+-        # Escape slashes and dots in paths for awk
```
Comment (pyhalov) [inline components/shell/bash-completion/patches/02-gawk.patch]: Generally, patch touches too much. Please, review it and remove unnecessary changes.

File: components/shell/bash-completion/patches/03-gsed.patch [hunk 3]
```diff
@@ -0,0 +1,2375 @@
+Overall bash-completion expects GNU sed.
+
+diff -pruNb bash-completion-2.8/bash_completion bash-completion-2.8.patched/bash_completion
+--- bash-completion-2.8/bash_completion	2018-05-16 23:47:09.344666760 +0000
++++ bash-completion-2.8.patched/bash_completion	2018-05-17 00:04:51.699094441 +0000
+@@ -111,7 +111,7 @@ _have()
+ }
+ 
+ # Backwards compatibility for compat completions that use have().
+-# @deprecated should no longer be used; generally not needed with dynamically
```
Comment (pyhalov) [inline components/shell/bash-completion/patches/03-gsed.patch]: Again, please, review this patch to avoid unnecessary changes.

**Comment by Mno-hime** [comment]
Thanks for the reviews, I prepared the patches more thoroughly this time. I used GNU tools only for completions which we either have in userland or which we at some point might have (e.g. `xfreerdp`), so we don't have to fix it when they are available.

I don't claim that every use of GNU utility in every case is proven to be necessary, but I have no means to verify that, it's just that I believe we are generally better with GNU utils then with illumos ones when it comes to bash-completion.

**Comment by pyhalov** [comment]
please, also add explicit dependencies on GNU grep,sed and awk to the package manifest
